merino/providers/adm.py

```diff
@@ -184,11 +185,18 @@ async def _fetch(self) -> None:
         for done_task in as_completed(fetch_tasks):
             res = await done_task
             for suggestion in res.json():
-                id = len(results)
-                for kw in suggestion.pop("keywords", []):
-                    # Note that for adM suggestions, each keyword can only be mapped to
-                    # a single suggestion.
-                    suggestions[kw] = id
+                kw_id = len(results)
+                keywords = suggestion.pop("keywords")
+                full_keywords = suggestion.pop("full_keywords")
+                begin = 0
+                for full_keyword, n in full_keywords:
+                    for query, fkw in zip(
+                        islice(keywords, begin, begin + n), repeat(full_keyword, n)
+                    ):
+                        # Note that for adM suggestions, each keyword can only be mapped to
+                        # a single suggestion.
+                        suggestions[query] = (kw_id, full_keyword)
+                        begin += n
```

Comment:
Incrementing `begin` within the inner loop would be catastrophic 😁 

```suggestion
                    begin += n
```

Reply:
eek

Reply:
Hmm, did this break anything in the contract tests? If not, I think we should add some tests to cover it.

Reply:
had the same thinking

Reply:
tests didn't pick up on this since all the tuples had values of 1, so the loop only go through once, changed contract test data for [this](https://github.com/mozilla-services/merino-py/pull/90/files#diff-39fd904b576c25a9c97637fd95124f3dab01b928e759b621930a33fde9d711d0R104-R107)

Reply:
This kind of logical coverage should be done in unit tests ideally. Could some be added?

Reply:
Yes, I think we can add some unit tests to `tests/unit/providers/test_adm.py` with some changes to the `FakeBackend` defined in there.

Reply:
okie doke, made some changes to `test_adm.py` for more realistic scenarios